tests/test_debugging.py

```diff
@@ -87,3 +88,19 @@ async def agen_func():
     await wait_all_tasks_blocked()
     await task
     await agen.aclose()
+
+
+async def test_wait_all_tasks_blocked_cancelled_task():
+    done = False
+
+    async def self_cancel(*, task_status):
+        task_status.started()
+        with move_on_after(-1):
+            await Event().wait()
+        nonlocal done
+        done = True
+
```

Comment:
More in line with the rest of the code base.
```suggestion
    async def self_cancel(*, task_status):
        nonlocal done
        task_status.started()
        with move_on_after(-1):
            await Event().wait()

        done = True

    done = False
```